### DISCUSSION TITLE
Custom div width not working with variables

### DISCUSSION THREAD

**Discussion by Duckbio** [tailwindlabs/tailwindcss#10666] [Help]
I am trying to make a progress bar using divs in Next JS and I've encountered a slight issue when setting a custom width; the width percentage doesn't work with variables unless they've been loaded manually. Here's the code: 
![image](https://user-images.githubusercontent.com/40851671/220944211-e44796eb-c9d5-41b7-821f-096a70268045.png)
![image](https://user-images.githubusercontent.com/40851671/220944746-ecb15540-4d94-479a-a002-4c06240e9e31.png)
This is the result. As shown, it works perfectly. However, if instead pass the percentage value as a variable it does not work.
![image](https://user-images.githubusercontent.com/40851671/220945229-a97953f8-ee75-4ffd-886a-7c53c4c7560a.png)
![image](https://user-images.githubusercontent.com/40851671/220945298-7aefc39c-2922-4468-9ffb-e7fddc470b65.png)
Interestingly, the class name is still present upon inspection
![image](https://user-images.githubusercontent.com/40851671/220945599-0a51295f-cf54-4e3f-8f83-0d22bf1b15dd.png)
It's just that it does not actually set a width; I've looked through all the computed styles and there is no width.
There is an exception though, if I set the variable to 10 (Which I loaded earlier without the variable) it works. The same effect happens if I repeat this with any value. For example, if I were to add "width-[40%]" and then change my variable to 40 and do "width-[" + width + "]" it would work, as well as the value 10 from earlier still working as well.
![image](https://user-images.githubusercontent.com/40851671/220946121-1d69b96e-5a83-4d2f-8abd-2404a58e737f.png)
![image](https://user-images.githubusercontent.com/40851671/220946191-b68bb28e-9c8a-4a7f-b0ae-c6eef892bbe2.png)

**Comment by Viscusier** [comment]
You have to make the variable a string, instead of an integer.

**Comment by zslabs** [comment]
Tailwind can't read your JS, you need to write full, static strings or have your selector reference a CSS variable for that width that you update elsewhere.
